### PR TITLE
Removing login functionality

### PR DESCRIPTION
**Pull request by nelsongallardo**
## Purpose :muscle:
Remove login functionality as this is going to be a reusable app, and login strategy should be part of another app or part of the project using this app
## Require changes (Database, config, etc)
In any local config file remove or comment out the authentication part.
```
REST_FRAMEWORK = {
    # TEMPORARILY COMMENTING AUTHENTICATION OUT
    # IN FUTURE CHANGES, AUTHENTICATION WILL BE REMOVED FOR THIS REUSABLE APP

    # 'DEFAULT_AUTHENTICATION_CLASSES': (
    #     'rest_framework.authentication.TokenAuthentication',
    # ),
    # 'DEFAULT_PERMISSION_CLASSES': (
    #     'rest_framework.permissions.IsAuthenticated',
    # ),
    'DEFAULT_FILTER_BACKENDS': ('django_filters.rest_framework.DjangoFilterBackend',)
}
```

### CODE REVIEW

File: api/admin.py [hunk 1]
```diff
@@ -1,3 +1,10 @@
...
+from django.apps import apps
 
-# Register your models here.
+models = apps.get_models()
+
+for model in models:
+    try:
+        admin.site.register(model)
+    except admin.sites.AlreadyRegistered:
+        pass
```
Comment (nelsongallardo) [inline api/admin.py:10]: Automatically registering all the models for django admin (the built-in CMS)

File: api/views/search_views.py [hunk 2]
```diff
@@ -13,7 +13,8 @@ def newest_mentors(request):
     """
     Retrieve the five newest mentors that joined, nesting user data
     """
-    mentor_profiles = MentorProfile.objects.select_related('user').filter(user__first_name__isnull=False, user__last_name__isnull=False, education__isnull=False, description__isnull=False, skills__isnull=False).order_by('-date_joined')[:30]
+
```
Comment (cesaraugustogarcia) [inline api/views/search_views.py]: Do we need this blank lines here?

**Review by cesaraugustogarcia — APPROVED** [review]
LGTM 🚀 
Just an small comment about some blank lines that may be removed, or maybe they are good for code readibility, your call.
Looking at this PR I noticed we should remove sendgrid stuff as well. Or create some interface to support mails feature but let the user implementing this reusable app to setup their own mail service.

**Comment by nelsongallardo** [comment]
> LGTM 🚀
> Just an small comment about some blank lines that may be removed, or maybe they are good for code readibility, your call.
> Looking at this PR I noticed we should remove sendgrid stuff as well. Or create some interface to support mails feature but let the user implementing this reusable app to setup their own mail service.

Yeah we should remove email logic, I'll create another PR for that. Instead of creating interfaces or coupling this app with email functionality, we should just send an event ("signal" is the event term in django) every time we were supposed to send an email. Then the project using this app will know what to do with that signal, whether it is triggering an email, a notification, or nothing.
Signals: https://docs.djangoproject.com/en/3.0/topics/signals/

**Comment by cesaraugustogarcia** [comment]
Let's merge this 💪 !

**Pull request merged**